go-controller/pkg/cni/cni.go

```diff
@@ -80,79 +78,70 @@ func (pr *PodRequest) cmdAdd() *PodResult {
 		annotation, err = kubecli.GetAnnotationsOnPod(namespace, podName)
 		if err != nil {
 			// TODO: check if err is non recoverable
-			logrus.Warningf("Error while obtaining pod annotations - %v", err)
+			logrus.Warningf("error getting pod annotations: %v", err)
 			return false, nil
 		}
 		if _, ok := annotation["ovn"]; ok {
 			return true, nil
 		}
 		return false, nil
 	}); err != nil {
-		logrus.Errorf("failed to get pod annotation - %v", err)
-		return nil
+		return nil, fmt.Errorf("failed to get pod annotation: %v", err)
 	}
 
 	ovnAnnotation, ok := annotation["ovn"]
 	if !ok {
```

Comment:
This can't actually happen; the wait loop will only exit without an error if `annotation["ovn"]` exists.
You could just make it so that `ovnAnnotation` is the variable passed out of the wait loop rather than `annotation`, and then it's obvious that you don't need the extra check here.